pkg/pillar/types/types.go

```diff
@@ -134,3 +134,19 @@ func FormatTriState(state TriState) string {
 	}
 	return ""
 }
+
+// ErrorInfo errorInfo holder structure
+type ErrorInfo struct {
+	Error       string
+	ErrorSource string
+	ErrorTime   time.Time
+}
+
+// NewErrorInfo - Create new instance of ErrorInfo
+func NewErrorInfo(err error, source string) ErrorInfo {
+	return ErrorInfo{
+		Error:       err.Error(),
+		ErrorSource: source,
+		ErrorTime:   time.Now(),
+	}
+}
```

Comment:
You only call this function in one place, it offers a very shallow abstraction (just syntax), yet the readers have to understand it. Oh, and it is very silently setting ErrorTime to now which is not what a potential future user would expect or want. Not a useful function. Please delete.

Reply:
I have to differ from you on this.

I expect ErrorInfo to become more ubiquitous.. At that time, we don't need the code to keep repeated at 100 differnt places - doing the same thing. There is no reason to have Error / ErrorTime repeated in various places when we have this structure in place. It is a very reasonable interface to create an error with current time set..

I want to retain this function..